inginious/common/course_user_setting.py

```diff
@@ -30,24 +30,38 @@ def __hash__(self):
 
     def get_id(self):
         """
-        :return: id of the additional field
+        :return: id of the setting
         """
         return self._id
 
     def get_description(self):
         """
-        :return: description of the additional field
+        :return: description of the setting
         """
         return self._description
 
     def get_type_name(self):
         """"
-        :return: type name of the additional field
+        :return: type name of the setting
         """
         return FieldTypes(self._type).name
 
     def get_type(self):
         """
-        :return: type of the additional field
+        :return: type of the setting
         """
         return self._type
+
+    def get_cast_type(self):
+        return FieldTypes(self._type).get_cast_type()
+
+    def render(self,value):
+        content = "<input id='" + self.get_id() + "' name='" + self.get_id() + "'"
```

Comment:
There is absolutely no additional value compared to the previous code. How is this piece of code more modular than the previous one ? Actually there even is a value loss as HTML code is formatted in Python.

What was meant was to have an abstract rendering method and a specific implementation for each kind of UserSetting.